broker/server.js

```diff
@@ -104,6 +114,39 @@ app.post("/Encounter", (req, res) => {
     .catch((e) => res.send(e));
 });
 
+app.post("/submit-vaccination", (req, res) => {
+  let imm = req.body.Immunization;
+  let obs = req.body.Observation;
+  let response = {
+    Immunization: "",
+    Observation: "",
+  };
+
+  // post resource
+  postImmunization(imm)
+    .then((imm_res) => {
+      response.Immunization = imm_res;
+
+      obs.partOf = `Immunization/${imm_res.id}`;
+      postObservation(obs)
+        .then((obs_res) => {
+          response.Observation = obs_res;
```

Comment:
Can you also get the reaction field of the Immunization to point to the newly created Observation? You can use a https://www.hl7.org/fhir/http.html#patch request to update a specific field in an existing resource.